lib/std/os.zig

```diff
@@ -254,6 +254,85 @@ pub fn close(fd: fd_t) void {
     }
 }
 
+pub const FChmodError = error{
+    AccessDenied,
+    InputOutput,
+    SymLinkLoop,
+    FileNotFound,
+    SystemResources,
+    ReadOnlyFileSystem,
+} || UnexpectedError;
+
+/// Changes the mode of the file referred to by the file descriptor.
+/// The process must have the correct privileges in order to do this
+/// successfully, or must have the effective user ID matching the owner
+/// of the file.
+pub fn fchmod(fd: fd_t, mode: mode_t) FChmodError!void {
+    if (builtin.os.tag == .windows or builtin.os.tag == .wasi)
+        @compileError("Unsupported OS");
+
+    while (true) {
+        const res = system.fchmod(fd, mode);
+
+        switch (system.getErrno(res)) {
+            .SUCCESS => return,
+            .INTR => continue,
+
+            .FAULT => unreachable,
```

Comment:
Rather than remove `BADF` and fall down to the `unexpectedErrno` case, I believe the convention is to handle known error codes with `unreachable`.  These 2 ways of handling the error have different affects.  In the latter case, it would prompt developers (like myself) to think that the std library is missing an error code and go in and add it.

Reply:
Oh, that was a complete mistake on my part, apologies for that. I'll fix that now 🤦